### PR TITLE
add laravel scout

### PR DESCRIPTION
**Pull request by igor-kamil**
implemented [laravel scout](https://laravel.com/docs/7.x/scout) using [elastic search driver](https://github.com/babenkoivan/scout-elasticsearch-driver)

- define indexes for `buildings` and `architects` 
- new deployment strategies (`scout_init` for first-run and `scout_update` for [zero-downtime es migration](https://github.com/babenkoivan/scout-elasticsearch-driver#zero-downtime-migration)
- accessor for `location_gps` attribute to unify the formatting of GPS coordinates (and to avoid exceptions while indexing)

### CODE REVIEW

File: app/Models/Building.php [hunk 1]
```diff
@@ -18,4 +116,25 @@ public function images()
...
+    	// convert Degree, Minutes, Seconds (DMS) to decimal if necessary
+    	if (strpos($value, '°') !== false) {
+    		$parts = preg_split("/[^\d\w.]+/", trim($value));
+    		$lat = $this->DMStoDD($parts[0], $parts[1], $parts[2], $parts[3]);
+		    $lng = $this->DMStoDD($parts[4], $parts[5], $parts[6], $parts[7]);
+		    return "$lat,$lng";
+    	}
+    }
+
+    private function DMStoDD($degrees, $minutes, $seconds, $direction)
```
Comment (eronisko) [inline app/Models/Building.php:131]: Would this fit better in the import job? That way we'd have clean data to start with and wouldn't have to create complex getters.

Comment (igor-kamil) [reply app/Models/Building.php:131]: you are right. I even created task for that -> https://jira.sng.sk/browse/REGARCH-27
this is more like 'hotfix" to make it works.

File: deployer/scout.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    run('{{bin/php}} {{release_path}}/artisan elastic:create-index "App\Elasticsearch\ArchitectsIndexConfigurator" -q');
+    run('{{bin/php}} {{release_path}}/artisan elastic:update-mapping "App\Models\Architect" -q');
+    run('{{bin/php}} {{release_path}}/artisan scout:import "\App\Models\Architect" -q');
+});
+
+desc('Update indexes mapping for Scout');
+task('scout:update', function () {
+	set('version_number', function () {
+	    return (string)time();
+	});
```
Comment (eronisko) [inline deployer/scout.php:19]: This works, just a suggestion. When we set this to a timestamp, it's hard to tell which version of the index you're actually running. If we kept a version number in a file, and load it from there every time, we could

1. tell which version of the index we're running (and compare them between environments), and 
1. prevent unnecessary migrations if we choose to.

File: deployer/scout.php [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+    run('{{bin/php}} {{release_path}}/artisan elastic:update-index "App\Elasticsearch\BuildingsIndexConfigurator" -q');
```
Comment (eronisko) [inline deployer/scout.php:21]: As mentioned earlier, please change this to something like `regarch:elastic:import`

File: app/Models/Building.php [hunk 4]
```diff
@@ -4,10 +4,108 @@
...
 class Building extends Model
 {
     use CrudTrait;
+    use Searchable;
+
+    protected $indexConfigurator = \App\Elasticsearch\BuildingsIndexConfigurator::class;
+
+    protected $searchRules = [
+        //
+    ];
```
[the remark this answers is not in the document]
Comment (igor-kamil) [reply app/Models/Building.php:18]: sooner or later we will add it anyway. https://github.com/babenkoivan/scout-elasticsearch-driver#search-rules
the default one works fine - but just for now.

File: config/scout_elastic.php [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+return [
+    'client' => [
+        'hosts' => [
+            env('SCOUT_ELASTIC_HOST', 'localhost:9200'),
+        ],
+    ],
+    'update_mapping' => env('SCOUT_ELASTIC_UPDATE_MAPPING', true),
+    'indexer' => env('SCOUT_ELASTIC_INDEXER', 'single'),
```
[the remark this answers is not in the document]
Comment (igor-kamil) [reply config/scout_elastic.php:10]: you are right. but for local development (debugging - if indexing has issues) - it's easier to keep `single`. I already set `SCOUT_ELASTIC_INDEXER=bulk` on production env file

**Comment by igor-kamil** [comment]
Thanks for comments. I like the idea with having command(s) to handle create / update elastic. 
I ended up with one command to cover both -> https://github.com/SlovakNationalGallery/register-architektury/pull/16/files#diff-c6cd0388d686477ef018b3507dac0178
The idea with storing index version in separate file - I suggest for keeping this for later.

**Pull request merged**